File: Stack.js
class Stack {
    constructor() {
        this.items = [];
        this.top = -1;
    }

    push(data) {
        this.items.push(data);
        this.top++;
    }

    pop() {
        if (this.top === -1) {
            return "Stack is empty";
        }
        this.items.pop();
        this.top--;
    }

    isEmpty() {
        return this.top === -1;
    }

    peek() {
        if (this.isEmpty()) {
            return "Stack is empty";
        }
        return this.items[this.top];
    }
}

const stack = new Stack();

stack.push(1);
stack.push(2);

console.log(stack);

File: Stack.java
public class Stack {
    int top;
    int[] items;

    Stack(int size) {
        top = -1;
        items = new int[size];
    }

    void push(int data) {
        if (top == items.length - 1) {
            System.out.println("Stack is full");
        } else {
            items[++top] = data;
        }
    }

    void pop() {
        if (top == -1) {
            System.out.println("Stack is empty");
        } else {
            System.out.println("Popped element: " + items[top--]);
        }
    }

    boolean isEmpty() {
        return top == -1;
    }

    int peek() {
        return items[top];
    }
}
